internal/services/authorization/parse/role_assignment.go

```diff
@@ -145,7 +136,15 @@ func RoleAssignmentID(input string) (*RoleAssignmentId, error) {
 		roleAssignmentId.Name = idParts[1]
 		roleAssignmentId.ManagementGroup = strings.TrimPrefix(idParts[0], "/providers/Microsoft.Management/managementGroups/")
 	default:
-		return nil, fmt.Errorf("could not parse Role Assignment ID %q", input)
+		re := regexp.MustCompile(`^/providers/(Microsoft.[a-zA-Z]+)/providers/Microsoft.Authorization/roleAssignments/([a-fA-F\d]{8}-[a-fA-F\d]{4}-[a-fA-F\d]{4}-[a-fA-F\d]{4}-[a-fA-F\d]{12})$`)
+		matches := re.FindStringSubmatch(input)
+		if len(matches) != 3 {
+			return nil, fmt.Errorf("could not parse Role Assignment ID %q", input)
+		}
+
+		roleAssignmentId.IsRootLevel = true
+		roleAssignmentId.ResourceProvider = matches[1]
+		roleAssignmentId.Name = matches[2]
```

Comment:
Can we avoid parsing with regular expressions and rather perform simple string matching/splitting? It's better that we test new RPs and explicitly support them.